src/GraphQL/Types/IResolveFieldContext.cs

```diff
@@ -0,0 +1,54 @@
+using GraphQL.Execution;
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+using System.Collections.Generic;
+using System.Threading;
+using Field = GraphQL.Language.AST.Field;
+
+namespace GraphQL.Types
+{
+    public interface IResolveFieldContext : IProvideUserContext
```

Comment:
I like that now all properties are read-only. Of course, someone could set them and his code will stop compiling, but this is not scary, since one can cast to class.

Reply:
Agreed